### PR TITLE
[fix](build index) fix core when build index for a new column which without data

### PR DESCRIPTION
**Pull request by Tanya-W**
## Proposed changes

Issue Number: close https://github.com/apache/doris/issues/27156

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Tanya-W** [comment]
run buildall

**Comment by Tanya-W** [comment]
run buildall

**Comment by Tanya-W** [comment]
run buildall

**Comment by hello-stephen** [comment]
run buildall

File: be/src/olap/task/index_builder.cpp [hunk 1]
```diff
@@ -198,6 +198,11 @@ Status IndexBuilder::handle_single_rowset(RowsetMetaSharedPtr output_rowset_meta
                 }
             }
 
+            if (return_columns.empty()) {
+                // no columns to read
+                break;
```
Comment (xiaokang) [inline be/src/olap/task/index_builder.cpp:203]: Do you mean continue? break will skip remaining segments that may have return_columns.

File: be/src/olap/task/index_builder.cpp [hunk 2]
```diff
@@ -280,6 +278,11 @@ Status IndexBuilder::_write_inverted_index_data(TabletSchemaSPtr tablet_schema,
         auto column = tablet_schema->column(column_idx);
         auto writer_sign = std::make_pair(segment_idx, index_id);
         std::unique_ptr<Field> field(FieldFactory::create(column));
+        auto converted_result = _olap_data_convertor->convert_column_data(i);
```
Comment (xiaokang) [inline be/src/olap/task/index_builder.cpp:281]: Is the problem avoid by using `_olap_data_convertor` after `if (column_idx < 0)` ?

File: be/src/olap/task/index_builder.cpp (hunk 1 above)
Comment (Tanya-W) [reply be/src/olap/task/index_builder.cpp:203]: return_columns is build by _alter_inverted_indexes, so the behavior of all segments is consistent, so here can be directly break

File: be/src/olap/task/index_builder.cpp (hunk 2 above)
Comment (Tanya-W) [reply be/src/olap/task/index_builder.cpp:281]: yes

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**